api/src/main/java/com/flipkart/poseidon/validator/TaskValidator.java

```diff
@@ -66,7 +72,11 @@
             for (Map.Entry<String, Object> contextEntry : context.entrySet()) {
                 final String key = contextEntry.getKey();
                 if (!datasourceRequestAttributes.contains(key)) {
-                    errors.add("ContextParam: " + braced(key) + " used in Task: " + braced(taskName) + " is not used in the Datasource");
+                    if (!skipUnusedFieldValidation) {
+                        errors.add("ContextParam: " + braced(key) + " used in Task: " + braced(taskName) + " is not used in the Datasource");
+                    }else {
+                        logger.info("Unused fields validation is disabled");
```

Comment:
Let's log the unused param and DS details as a warning. Something like 
logger.warn("ContextParam: {} used in Task: {} is not used in the Datasource", braced(key), braced(taskName));